llvm/test/MC/ARM/macho-relocs-with-addend-invalid.s

```diff
@@ -0,0 +1,28 @@
+// RUN: not llvm-mc -triple armv7-apple-darwin -filetype=obj %s 2>&1 | FileCheck %s
+
+_foo:
+    // Check that the relocation size is valid.
+
+    // CHECK: :[[@LINE+1]]:{{[0-9]+}}: error: Relocation out of range
+    bl  _foo+0xfffffff00
```

Comment:
This looks like a 36bit address? That's pretty big! There are some tests of what GCC uses as the limits in https://godbolt.org/z/n4drer68h, it would be good to test those edge cases.

Reply:
I'll add them and modify the test case here, thanks.

Reply:
There's one edge case there that I think I didn't understand correctly:

_foo2:
    bl  _foo2-0x1fffffc

If I multiple the number by -1, in 2's complement I get 0x2000004, so this should be the same as:

_foo2:
   bl _foo2+0x2000004

Which gcc does accept (and it should, it's a legal offset). What am I missing here?

Reply:
Hi. I think you should treat the value as a 32bit integer, in which case multiplying by -1 would give 0xFE000004.

Reply:
When I truncate that to 26 bits, I still get 0x2000004, which is a valid addend (according to a test case directly below that). This behavior seems inconsistent. I did add a similar test now, but I use -0x2000010

Reply:
The final instruction will take the value as a signed number and add/subtract it from a 32bit address (the current PC), so in that regard there is no wrapping at 26bits. I believe (mostly trusting what GCC is doing) that the valid range is between -0x1fffff8 and +0x2000004.

Reply:
Aligned now to the GCC behavior.